src/main/java/configurationslicing/docker/AbstractDockerSlicerSpec.java

```diff
@@ -0,0 +1,177 @@
+package configurationslicing.docker;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.cloudbees.dockerpublish.DockerBuilder;
+import com.google.common.base.Throwables;
+import com.google.common.collect.Lists;
+import configurationslicing.UnorderedStringSlicer.UnorderedStringSlicerSpec;
+import hudson.matrix.MatrixProject;
+import hudson.model.AbstractProject;
+import hudson.model.Descriptor;
+import hudson.model.Project;
+import hudson.tasks.Builder;
+import hudson.util.DescribableList;
+import jenkins.model.Jenkins;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class AbstractDockerSlicerSpec
+        extends UnorderedStringSlicerSpec<AbstractProject<?, ?>> {
+
+    private static final Logger log = LoggerFactory.getLogger(AbstractDockerSlicerSpec.class);
+
+    private static final String NOTHING = "(Nothing)";
+    static final String DEFAULT = "(Default)";
+
+    public abstract String getName();
+    public abstract String getUrl();
+    public abstract String getSliceParam(DockerBuilder builder);
+    public abstract DockerBuilder setSliceParam(DockerBuilder builder, String value);
+
+    @Override
+    public String getDefaultValueString() {
+        return DEFAULT;
+    }
+
+    @Override
+    public String getName(AbstractProject<?, ?> item) {
+        return item.getDisplayName();
+    }
+
+    public List<AbstractProject<?, ?>> getWorkDomain() {
+        List<AbstractProject<?, ?>> list = new ArrayList<AbstractProject<?, ?>>();
+        List<AbstractProject> temp = Jenkins.getInstance().getAllItems(AbstractProject.class);
+        for (AbstractProject p: temp) {
```

Comment:
`list`, `temp` and `p` are not very descriptive. They don't explain what each of them do so it's hard to understand what the method's purpose is.